synapse/storage/schema/delta/32/remove_indices.sql

```diff
@@ -16,27 +16,24 @@
 
 -- The following indices are redundant, other indices are equivalent or
 -- supersets
-DROP INDEX IF EXISTS events_room_id;
-DROP INDEX IF EXISTS events_order;
-DROP INDEX IF EXISTS events_topological_ordering;
-DROP INDEX IF EXISTS events_stream_ordering;
-DROP INDEX IF EXISTS state_groups_id;
-DROP INDEX IF EXISTS event_to_state_groups_id;
-DROP INDEX IF EXISTS event_push_actions_room_id_event_id_user_id_profile_tag;
-DROP INDEX IF EXISTS event_push_actions_room_id_user_id;
-
-DROP INDEX IF EXISTS event_destinations_id;
-DROP INDEX IF EXISTS st_extrem_id;
-DROP INDEX IF EXISTS event_content_hashes_id;
-DROP INDEX IF EXISTS event_signatures_id;
-DROP INDEX IF EXISTS event_edge_hashes_id;
-DROP INDEX IF EXISTS redactions_event_id;
-DROP INDEX IF EXISTS remote_media_cache_thumbnails_media_id;
-DROP INDEX IF EXISTS room_hosts_room_id;
-DROP INDEX IF EXISTS event_search_ev_ridx;
+DROP INDEX IF EXISTS events_room_id; -- Prefix of events_room_stream
+DROP INDEX IF EXISTS events_order; -- Prefix of events_order_topo_stream_room
+DROP INDEX IF EXISTS events_topological_ordering; -- Prefix of events_order_topo_stream_room
+DROP INDEX IF EXISTS events_stream_ordering; -- Duplicate of PRIMARY KEY
+DROP INDEX IF EXISTS state_groups_id; -- Duplicate of PRIMARY KEY
+DROP INDEX IF EXISTS event_to_state_groups_id; -- Duplicate of PRIMARY KEY
+DROP INDEX IF EXISTS event_push_actions_room_id_event_id_user_id_profile_tag; -- Duplicate of UNIQUE CONSTRAINT
 
+DROP INDEX IF EXISTS event_destinations_id; -- Prefix of UNIQUE CONSTRAINT
+DROP INDEX IF EXISTS st_extrem_id; -- Prefix of UNIQUE CONSTRAINT
+DROP INDEX IF EXISTS event_content_hashes_id; -- Prefix of UNIQUE CONSTRAINT
+DROP INDEX IF EXISTS event_signatures_id; -- Prefix of UNIQUE CONSTRAINT
+DROP INDEX IF EXISTS event_edge_hashes_id; -- Prefix of UNIQUE CONSTRAINT
+DROP INDEX IF EXISTS redactions_event_id; -- Duplicate of UNIQUE CONSTRAINT
+DROP INDEX IF EXISTS room_hosts_room_id; -- Prefix of UNIQUE CONSTRAINT
 
 -- The following indices were unused
+DROP INDEX IF EXISTS remote_media_cache_thumbnails_media_id;
 DROP INDEX IF EXISTS evauth_edges_auth_id;
 DROP INDEX IF EXISTS topics_room_id;
```

Comment:
I think that either I'd want to keep this index or get rid of the entire table given that it is write only in synapse, and can be fairly easily constructed from the state tables for ad-hoc queries.